fenix/app/src/main/java/org/mozilla/fenix/addons/ui/AddonPermissionsScreen.kt

```diff
@@ -0,0 +1,75 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.addons.ui
+
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import org.mozilla.fenix.R
+import org.mozilla.fenix.compose.LinkText
+import org.mozilla.fenix.compose.LinkTextState
+import org.mozilla.fenix.theme.FirefoxTheme
+
+private const val LEARN_MORE_URL =
+    "https://support.mozilla.org/kb/permission-request-messages-firefox-extensions"
+
+/**
+ * We create the Addon permissions screen by adding each section
+ */
+@Composable
+fun AddonPermissionsScreen(
+    permissions: List<String>,
+    optionalPermissions: List<String>,
+    websitesWithPermissions: List<String>,
+) {
+    val state = LinkTextState(
+        text = stringResource(R.string.mozac_feature_addons_learn_more),
+        url = LEARN_MORE_URL,
+        onClick = {},
+    )
+
+    Column(modifier = Modifier.padding(vertical = 8.dp)) {
+        SectionRequired(permissions)
+
+        if (optionalPermissions.isNotEmpty()) {
+            SectionOptional(optionalPermissions)
+        }
+
+        SectionModifyWebsiteData(websitesWithPermissions)
+
+        Row(
+            modifier = Modifier
+                .fillMaxWidth()
+                .padding(16.dp),
+        ) {
+            LinkText(
+                text = stringResource(R.string.mozac_feature_addons_learn_more),
+                linkTextStates = listOf(state),
+            )
+        }
+    }
+}
+
+@Composable
+@Preview(name = "Addon permissions screen")
+fun AddonPermissionsScreenPreview() {
+    val permissions: List<String> = listOf("Permission required 1", "Permission required 2")
+    val optionalPermissions: List<String> = listOf("Optional permission 1")
+    val websitesWithPermissions: List<String> = listOf(
+        "https://required.website",
+        "https://optional-suggested.website...",
+        "https://user-added.website.com",
+    )
```

Comment:
You could take a look to this data structure as this is the way the API will looks like https://github.com/mozilla-mobile/firefox-android/pull/5793/files#diff-26e01436aa717b89aba5b097f97aca6777693d162bcf030f28006165c6998414R64

Reply:
What's the context for this comment? This is a preview so not sure what to do with it.